plugins/module_utils/linode_event_poller.py

```diff
@@ -0,0 +1,114 @@
+"""
+Contains a helper class for polling for resource events.
+"""
+
+from typing import Dict, Any
+
+import polling
+from linode_api4 import LinodeClient, Event
+
+from ansible_collections.linode.cloud.plugins.module_utils.linode_timeout import TimeoutContext
+
+
+POLL_INTERVAL_SECONDS = 2
+
+
+class EventPoller:
+    """
+    EventPoller allows modules to dynamically poll for Linode events
+    """
+
+    def __init__(self, client: LinodeClient, entity_type: str, action: str, entity_id: int = 0):
+        self._client = client
+        self._entity_type = entity_type
+        self._entity_id = entity_id
+        self._action = action
+
+        # Initialize with an empty cache if no entity is specified
+        if self._entity_id == 0:
+            self._previous_event_cache = {}
+            return
+
+        # We only want the first page of this response
+        result = client.get('/account/events', filters=self._build_filter())
+
+        self._previous_event_cache = {v['id']: v for v in result['data']}
+
+    def _build_filter(self) -> Dict[str, Any]:
+        """Generates a filter dict to use in HTTP requests"""
+        return {
+            '+order': 'desc',
+            '+order_by': 'created',
+            'entity.id': self._entity_id,
+            'entity.type': self._entity_type,
+            'action': self._action
+        }
+
+    def set_entity_id(self, entity_id: int) -> None:
+        """
+        Sets the ID of the entity to filter on.
+        This is useful for create operations where
+        the entity id might not be known in __init__.
+        """
+        self._entity_id = entity_id
+
+    def wait_for_next_event(self, timeout: int) -> Event:
+        """
+        Waits for and returns the next event matching the
+        poller's configuration.
+        """
+        result_event: Dict[str, Any] = {}
+
+        def poll_func():
+            result = self._client.get('/account/events', filters=self._build_filter())
+            for event in result['data']:
+                if event['id'] in self._previous_event_cache:
+                    continue
+
+                nonlocal result_event
+                result_event = event
+
+                # Merge the new events into the cache
+                for event in result['data']:
+                    if event['id'] in self._previous_event_cache:
+                        continue
+
+                    self._previous_event_cache[event['id']] = event
```

Comment:
This loop seems a duplicated of the loop at line 64. Is it possible to merge and simplified the logic of them?

Reply:
Something in my brain looks like:
```python
        def poll_func():
            result = self._client.get('/account/events', filters=self._build_filter())
            event_found = False
            for event in result['data']:
                if event['id'] in self._previous_event_cache:
                    continue

                nonlocal result_event
                if not event_found:
                    result_event = event
                    event_found = True
                self._previous_event_cache[event['id']] = event

            return event_found
```

Reply:
Hmm, come to think of it the second loop seems redundant anyway. Lemme fix that